apps/recnet-api/src/modules/invite-code/invite-code.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Inject, Injectable } from "@nestjs/common";
+import Chance from "chance";
+
+import InviteCodeRepository from "@recnet-api/database/repository/invite-code.repository";
+import {
+  InviteCode as DbInviteCode,
+  InviteCodeFilterBy,
+} from "@recnet-api/database/repository/invite-code.repository.type";
+import { getOffset } from "@recnet-api/utils";
+
+import { InviteCode } from "./entities/invite-code.entity";
+import {
+  CreateInviteCodeResponse,
+  GetInviteCodeResponse,
+} from "./invite-code.response";
+
+@Injectable()
+export class InviteCodeService {
+  constructor(
+    @Inject(InviteCodeRepository)
+    private readonly inviteCodeRepository: InviteCodeRepository
+  ) {}
+
+  public async createInviteCode(
+    numCodes: number,
+    ownerId: string
+  ): Promise<CreateInviteCodeResponse> {
+    const codes = Array.from({ length: numCodes }, () => this.genRandomCode());
```

Comment:
Yes, it's possible. But asking this question is like asking is it possible that hash function have collision lol. Yes, hash function might have collision. But the possibility is sufficiently low. (Maybe it won't happen in our life-time.)

I did not add the check since I think the possibility is low enough (our invite-code consists 16 numeric or lowercase characters. So the possibility would be `1/(36)^16`). I guess it's low enough.

And I think adding check here would be weird(?). If I'm going to check every time after I generate an invite-code. Is it going to be a performance issue to check every time (idk, let's say we are going to generate 20 codes. Would it be slow?).

I think if we still have concern about collision, the better way to solve this issue is switch to more robust invite-code-generation function (maybe use hash function?).

Reply:
The `createMany` operation is atomic. If any code collides with the existing code in DB, it will throw a "unique constraint failed" error from prisma and none of the codes will be inserted. The probability will grow with the total count of the invite codes. I understand that even if it grows to 10^5/36^16, it's still small. But I'll suggest at least adding wording such as "please try again" on the front-end side error handling in case people encounter this issue.